hibernate-cache/src/main/java/org/hibernate/cache/infinispan/access/TxInvalidationInterceptor.java

```diff
@@ -119,15 +127,17 @@ public Object visitLockControlCommand(TxInvocationContext ctx, LockControlComman
 	}
 
 	private Object handleInvalidate(InvocationContext ctx, WriteCommand command, Object... keys) throws Throwable {
-		Object retval = invokeNextInterceptor( ctx, command );
-		if ( command.isSuccessful() && !ctx.isInTxScope() ) {
-			if ( keys != null && keys.length != 0 ) {
-				if ( !isLocalModeForced( command ) ) {
-					invalidateAcrossCluster( isSynchronous( command ), keys, ctx );
-				}
-			}
-		}
-		return retval;
+      return invokeNextAndHandle( ctx, command, (rCtx, rCommand, rv, throwable) -> {
```

Comment:
I'm not sure you can do this one, since the handler depends on the `keys` parameter. If it's capturing an external variable, a new lambda needs to be created each time.